lib/blogs_api/user.ex

```diff
@@ -8,28 +8,39 @@ defmodule BlogsApi.User do
   @primary_key {:id, Ecto.UUID, autogenerate: true}
 
   schema "users" do
-    field :display_name, :string
-    field :email, :string
-    field :password, :string
-    field :image, :string
+    field(:display_name, :string)
+    field(:email, :string)
+    field(:encrypted_password, :string)
+    field(:password, :string, virtual: true)
+    field(:image, :string)
+  end
+
+  @required_params ~w(display_name email password image)a
+  def build(params) do
+    params
+    |> changeset()
+    |> apply_action(:insert)
   end
 
-  @required_params ~w(display_name email password)a
   def changeset(params) do
     %__MODULE__{}
     |> cast(params, @required_params)
     |> validate_required(@required_params)
     |> validate_required(:email, message: "\"email\" is required")
     |> validate_required(:password, message: "\"password\" is required")
```

Comment:
Acredito que não é necessário colocar `:email` e `:password` separadamente para serem validados como obrigatórios, já que eles estão entre os `@required_parameters` que já estão sendo validados.

```suggestion
    |> validate_required(@required_params)
```